module/behaviour/planning/KickPlanner/src/KickPlanner.cpp

```diff
@@ -131,14 +125,14 @@ namespace module::behaviour::planning {
                     switch (kick_plan.kick_type.value) {
                         case KickType::IK_KICK:
                             if (ball_position.y() > 0.0) {
-                                emit(std::make_unique<KickCommand>(KickCommand(Eigen::Vector3d(0.1, 0.04, 0.0),
-                                                                               Eigen::Vector3d::UnitX(),
+                                emit(std::make_unique<KickCommand>(KickCommand(Eigen::Vector3f(0.1, 0.04, 0.0),
```

Comment:
`0.1` has type `double`, `0.1f` has type `float`. Prevent implicit casts and make things explicit (do this everywhere)
```suggestion
                                emit(std::make_unique<KickCommand>(KickCommand(Eigen::Vector3f(0.1f, 0.04f, 0.0f),
```